src/Views/Calendar/CalendarView.js

```diff
@@ -0,0 +1,530 @@
+    const view = App.getView(detailView);
+    descriptor = (entry.isEvent) ? descriptor : entry.Description;
+    if (view) {
+      view.show({
+        title: descriptor,
+        key: key,
+      });
+    }
+  },
+  navigateToInsertView: function navigateToInsertView() {
+    const view = App.getView(this.insertView || this.editView);
+
+    if (!this.options) {
+      this.options = {};
+    }
+
+    this.options.currentDate = this.currentDate.toString('yyyy-MM-dd') || moment().startOf('day');
+    if (view) {
+      view.show({
+        negateHistory: true,
+        returnTo: this.id,
+        insert: true,
+        currentDate: this.options.currentDate.valueOf(),
+      });
+    }
+  },
+  parseName: function parseName(name = {}) {
+    return name.split(' ').splice(-1)[0];
+  },
+  process: function process(store, entries) {
+    const count = entries.length;
+
+    if (count > 0) {
+      for (let i = 0; i < count; i++) {
+        const entry = this._processEntry(entries[i]);
+        // If key comes back with nothing, check that the store is properly
+        // setup with an idProperty
+        const entryKey = store.getIdentity(entry);
+        this.entries[entryKey] = entry;
+        const date = moment(convert.toDateFromString(entry.StartDate)).format('YYYY-MM-DD');
+        if (this.monthActivities[date]) {
+          this.monthActivities[date].push(entryKey);
+        } else {
+          this.monthActivities[date] = [entryKey];
+        }
+      }
+    }
+  },
+  processData: function processData(entries) {
+    if (!entries) {
+      return;
+    }
+
+    const store = this.get('store');
+
+    this.process(store, entries);
+  },
+  processEventData: function processEventData(entries) {
+    if (!entries) {
+      return;
+    }
+    const store = this.createEventStore();
+
+    this.process(store, entries);
+  },
+  processShowOptions: function processShowOptions(options) {
+    if (options.currentDate) {
+      this.currentDate = moment(options.currentDate).startOf('day') || moment().clone().startOf('day');
+      this.refreshRequired = true;
+    }
+  },
+  refresh: function refresh() {
+    this.renderCalendar();
+  },
+  refreshData: function refreshData() {
+    this._dataLoaded = false;
+    domClass.add(this.activityContainerNode, 'list-loading');
+    this.set('activityContent', this.loadingTemplate.apply(this));
+    this.currentDate = this._calendar.getSelectedDateMoment();
+    this.queryText = '';
+    this.monthActivities = [];
+    this.activityQuery = this.formatQueryActivity(this.currentDate);
+    this.eventQuery = this.formatQueryEvent(this.currentDate);
+    const activityPromise = new Promise((resolve, reject) => { // eslint-disable-line
+      resolve(this.requestData());
+    });
+    const eventPromise = new Promise((resolve, reject) => { // eslint-disable-line
+      resolve(this.requestEventData());
+    });
+    Promise.all([activityPromise, eventPromise]).then(() => {
```

Comment:
`this.requestData()` and `this.requestEventData()` already return a deferred/promise. Promise.all shouldn't need the wrappers above it (activityPromise and eventPromise). It should be able to just do `Promise.all([this.requestData(), this.requestEventData()]).then(...)`